ASTProcessor/src/main/java/edu/usb/argos/ASTProcessor/reader/validations/DirectoryPathAnalyzer.java

```diff
@@ -0,0 +1,31 @@
+package edu.usb.argos.ASTProcessor.reader.validations;
+
+import edu.usb.argos.ASTProcessor.reader.interfaces.IPathValidator;
+import org.springframework.stereotype.Component;
+
+import java.nio.file.Files;
+import java.nio.file.NoSuchFileException;
+import java.nio.file.Path;
+
+@Component
+public class DirectoryPathAnalyzer implements IPathValidator {
+
+    @Override
+    public void validatePath(Path path) throws Exception {
+        existsDirectoryPath(path);
+        isDirectoryPath(path);
+    }
+
+    private void existsDirectoryPath(Path directoryPath) throws NoSuchFileException {
+        if (!Files.exists(directoryPath)) {
+            throw new NoSuchFileException("No found directory for: " + directoryPath);
+        }
+    }
+
+    private void isDirectoryPath(Path directoryPath) throws IllegalArgumentException {
+        if (!Files.isDirectory(directoryPath)) {
+            throw new IllegalArgumentException("The path provided is not a directory: " + directoryPath);
```

Comment:
same here use use @Slf4j to log exceptions

Reply:
done